q2_amr/card/partition.py

```diff
@@ -149,37 +161,53 @@ def _partition_annotations(
         )
         num_partitions = num_annotations
 
-    # Split the dict into a list of specified number of dicts
-    i = itertools.cycle(range(num_partitions))
-    annotations_split_list = [{} for _ in range(num_partitions)]
-    for key, value in annotations_dict.items():
-        annotations_split_list[next(i)][key] = value
+    # Splits annotations into the specified number of sublists
+    annotations_all_split = split_list(annotations_all, num_partitions)
```

Comment:
Can't you just use the `np.array_split` like we do in every other partitioning action?

Reply:
Hi @misialq 
I tried to use the array_split function of numpy but because my tuples have multiple paths in them and not only one path like in the other partition methods, it does not work.
I get this error: ValueError: setting an array element with a sequence. The requested array has an inhomogeneous shape after 2 dimensions. The detected shape was (3, 3) + inhomogeneous part. And i couldn't get it to work.

I have now created a small helper function called split list that splits the list into a number of sublists.

Reply:
That happens because numpy is trying to convert your inner lists into another dimension. You can just do `np.array_split(np.array(annotations_all, dtype=object), num_partitions)` and it should hopefully achieve exactly the same.